spec/bolt_server/app_integration_spec.rb

```diff
@@ -171,6 +176,191 @@ def app
+        target = {
+          hostname: puppet_6_agent_container.host,
+          user: puppet_6_agent_container.user,
+          password: puppet_6_agent_container.password,
+          port: puppet_6_agent_container.port
+        }
+        body = {
+          versioned_project: 'bolt_server_test_project',
+          target: target,
+          job_id: 1
+        }
+        post(path, JSON.generate(body), 'CONTENT_TYPE' => 'text/json')
+        expect(last_response.status).to eq(400)
+        result = JSON.parse(last_response.body)
+        expect(result['kind']).to eq('bolt-server/request-error')
+        expect(result['msg']).to eq("Target must have 'task' plugin hook")
+      end
+    end
+
+    describe "apply" do
+      def cross_platform_catalog(certname)
+        {
+          "catalog" => {
+            "tags" => [
+              "settings"
+            ],
+            "name" => certname,
+            "version" => 1581636379,
+            "code_id" => nil,
+            "catalog_uuid" => "5a4372c6-253f-46df-be99-3c40c9922423",
+            "catalog_format" => 1,
+            "environment" => 'bolt_catalog',
+            "resources" => [
+              {
+                "type" => "Stage",
+                "title" => "main",
+                "tags" => %w[
+                  stage
+                  class
+                ],
+                "exported" => false,
+                "parameters" => {
+                  "name" => "main"
+                }
+              },
+              {
+                "type" => "Class",
+                "title" => "Settings",
+                "tags" => %w[
+                  class
+                  settings
+                ],
+                "exported" => false
+              },
+              {
+                "type" => "Class",
+                "title" => "main",
+                "tags" => [
+                  "class"
+                ],
+                "exported" => false,
+                "parameters" => {
+                  "name" => "main"
+                }
+              },
+              {
+                "type" => "Notify",
+                "title" => "hello world",
+                "tags" => %w[
+                  notify
+                  class
+                ],
+                "line" => 1,
+                "exported" => false
+              }
+            ],
+            "edges" => [
+              {
+                "source" => "Stage[main]",
+                "target" => "Class[Settings]"
+              },
+              {
+                "source" => "Stage[main]",
+                "target" => "Class[main]"
+              },
+              {
+                "source" => "Class[main]",
+                "target" => "Notify[hello world]"
+              }
+            ],
+            "classes" => [
+              "settings"
+            ]
+          }
+        }
+      end
+
+      it 'applies a catalog' do
```

Comment:
Do we want a test to see what this does if the catalog fails to apply?

Reply:
I dont think we have anything at this level valuable to test. I have coverage for this case in orch https://github.com/puppetlabs/orchestrator/pull/1778/files#diff-5c92a78fec2c5e883d5e84edbbd419974aaf1a39ace4f4464dfa0a5720469d41R109-R117